app/src/main/java/org/jellyfin/androidtv/util/RecommendedServerIssueExtensions.kt

```diff
@@ -47,12 +50,15 @@ fun RecommendedServerIssue.getFriendlyMessage(context: Context): String = when (
  */
 @Suppress("MagicNumber")
 fun Collection<RecommendedServerIssue>.getSummary(context: Context): String? = maxByOrNull {
-	// Assign a "score" to each issue type so we can return the most important one
+	// Assign a score to each issue type so we can return the most important one. Higher number
+	// means more important.
 	when (it) {
-		is RecommendedServerIssue.MissingSystemInfo -> 5
-		is RecommendedServerIssue.UnsupportedServerVersion -> 4
-		is RecommendedServerIssue.OutdatedServerVersion -> 3
-		is RecommendedServerIssue.SlowResponse -> 2
+		is RecommendedServerIssue.MissingSystemInfo -> 7
+		is RecommendedServerIssue.UnsupportedServerVersion -> 6
+		is RecommendedServerIssue.OutdatedServerVersion -> 5
+		is RecommendedServerIssue.SecureConnectionFailed -> 4
+		is RecommendedServerIssue.SlowResponse -> 3
+		is RecommendedServerIssue.ServerUnreachable -> 2
 		RecommendedServerIssue.MissingVersion -> 1
```

Comment:
How does this work without the `is`?

Reply:
MissingVersion is an object so we can use an equals check. The others are data classes so we need to check if it is an instance of that class.

Reply:
See https://github.com/jellyfin/jellyfin-sdk-kotlin/blob/master/jellyfin-core/src/commonMain/kotlin/org/jellyfin/sdk/discovery/RecommendedServerIssue.kt#L34